apps/storybook/src/Utilities.stories.mdx

```diff
@@ -163,12 +163,23 @@ const { xVisibleDomain, yVisibleDomain } = useVisibleDomains();
 
 #### useFrameRendering
 
-Trigger a re-render on every React Three Fiber frame when panning/zooming. Avoid calling this hook in a component that is slow to render to ensure pan/zoom interactions remain responsive.
+Trigger a re-render on every React Three Fiber frame when panning/zooming. Avoid calling this hook in a component that is slow to render to ensure pan/zoom interactions remain responsive. Must be called from a child component of `VisCanvas`.
 
 ```ts
 useFrameRendering(): void
 ```
 
+#### useCanvasEvents
+
+Register event listeners on the `canvas` element. Useful to implement custom interactions. Must be called from a child component of `VisCanvas`.
+
+```ts
+useCanvasEvents(callbacks: CanvasEventCallbacks): void
+
+const handlePointerDown = useCallback((evt: PointerEvent) => { ... }, []);
+useCanvasEvents({ onPointerDown: handlePointerDown }); // also supported: `onPointerMove`, `onPointerUp` and `onWheel`
+```
+
 ### AxisSystemContext
 
 Children of `VisCanvas` have access to the `AxisSystemContext`, which exposes the visualization's size, the axes' configurations and scales, as well as two utilities to transform a vector from world to data space and from data to world space:
```

Comment:
We could add a part about the new utility to transform a vector from world position to HTML position (top, left).

Given that the list is becoming bigger, we could make a table like https://github.com/pmndrs/react-three-fiber/blob/78046f254f6bbacec6c2d3a62f0d0f673ed5fc52/docs/API/hooks.mdx#state-properties

Reply:
agreed!